### PR TITLE
Restore Component Image controller

### PR DESCRIPTION
**Pull request by mmorhun**
Signed-off-by: Mykola Morhun <mmorhun@redhat.com>

Reworks Component Image controller to search for newly built image in pipeline results instead of build task results.

### CODE REVIEW

File: controllers/component_image_controller.go [hunk 1]
```diff
@@ -295,6 +257,41 @@ func (r *ComponentImageReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	return ctrl.Result{}, nil
 }
 
+func getImageReferenceFromBuildPipeline(buildPipelineRun tektonapi.PipelineRun) (string, error) {
+	pipelineResults := buildPipelineRun.Status.PipelineResults
```
Comment (gabemontero) [inline controllers/component_image_controller.go:261]: ok cool you sorted out your "how to get pipeline results from status" question you had yesterday :-)

File: controllers/component_image_controller.go [hunk 2]
```diff
@@ -295,6 +257,41 @@ func (r *ComponentImageReconciler) Reconcile(ctx context.Context, req ctrl.Reque
...
+	if pipelineResults == nil {
+		return "", fmt.Errorf("PipelineRun.Status.PipelineResults is empty")
+	}
+
+	imageUrl := ""
+	for _, pipelineRunResult := range pipelineResults {
+		if pipelineRunResult.Name == "IMAGE_URL" {
```
Comment (gabemontero) [inline controllers/component_image_controller.go:268]: who is creating theses pipelineruns @mmorhun , build-service or application-service?  

are constants for these names not defined in one of the repos?  or maybe the new application-api repo?

**Review by gabemontero — APPROVED** [review]
LGTM

one minor comment / not a blocker per se but I'm curious

File: controllers/component_image_controller.go (hunk 2 above)
Comment (mmorhun) [reply controllers/component_image_controller.go:268]: Initial build pipeline is created by build-service. Next are created by Tekton or PaC.
The constants defined [here](https://github.com/redhat-appstudio/build-definitions/blob/main/pipelines/base/template-build/template-build.yaml#L99-L102).

**Review by Michkov — APPROVED** [review]

**Pull request merged**